### PR TITLE
VACCINECER-974 generation api character validation

### PR DESCRIPTION
**Pull request by Fabiansson**
generation API now validates against most common special characters. Also invalidates simple html script tags.

### CODE REVIEW

File: src/main/java/ch/admin/bag/covidcertificate/api/request/CovidCertificatePersonNameDto.java [hunk 1]
```diff
@@ -17,11 +20,24 @@
     private String givenName;
 
     public void validate() {
+        Pattern p = Pattern.compile("[!@#\\r\\n$%¶*\\\\()_:/+=|<>?{}\\[\\]~-]");
```
Comment (Armin-Isenring-Bit) [inline src/main/java/ch/admin/bag/covidcertificate/api/request/CovidCertificatePersonNameDto.java:23]: I'd upgrade this attribute to a class property. The pattern compiling is counting to the method runtime, though it has no dynamic aspects and may be defined statically.

File: src/main/java/ch/admin/bag/covidcertificate/api/request/CovidCertificatePersonNameDto.java [hunk 2]
```diff
@@ -17,11 +20,24 @@
...
+
         if (!StringUtils.hasText(givenName) || givenName.length() > MAX_STRING_LENGTH) {
             throw new CreateCertificateException(INVALID_GIVEN_NAME);
         }
         if (!StringUtils.hasText(familyName) || familyName.length() > MAX_STRING_LENGTH) {
             throw new CreateCertificateException(INVALID_FAMILY_NAME);
         }
+
+        Matcher givenNameMatcher = p.matcher(givenName);
+        Matcher familyNameMatcher = p.matcher(familyName);
```
Comment (Armin-Isenring-Bit) [inline src/main/java/ch/admin/bag/covidcertificate/api/request/CovidCertificatePersonNameDto.java:34]: Put the creation of the familyMatcher after the if of givenNameMatcher (before line 39). If the givenNameMatcher.find() succeeds the familyNameMatcher is not needed anymore.

**Review by Armin-Isenring-Bit — CHANGES_REQUESTED** [review]
Changing pattern.compile attribute has a benefit, the order of Matcher creation is a matter of taste, imho.

File: src/main/java/ch/admin/bag/covidcertificate/api/request/CovidCertificatePersonNameDto.java (hunk 1 above)
Comment (Fabiansson) [reply src/main/java/ch/admin/bag/covidcertificate/api/request/CovidCertificatePersonNameDto.java:23]: @Armin-Isenring-Bit  But should we have any other class properties than business-properties in a Dto?

Comment (Armin-Isenring-Bit) [reply src/main/java/ch/admin/bag/covidcertificate/api/request/CovidCertificatePersonNameDto.java:25]: Yeah, but in this case I'd do it as static final which should be not counted towards a dto property. But then again, I might be wrong...

Comment (Fabiansson) [reply src/main/java/ch/admin/bag/covidcertificate/api/request/CovidCertificatePersonNameDto.java]: Upgraded to static final class property now

**Review by Armin-Isenring-Bit — APPROVED** [review]

**Pull request merged**